featureflags.go

```diff
@@ -142,43 +202,382 @@ func (poller *FeatureFlagsPoller) IsFeatureEnabled(key string, distinctId string
+		return !strings.Contains(strings.ToLower(fmt.Sprintf("%v", override_value)), strings.ToLower(fmt.Sprintf("%v", value))), nil
+	}
+
+	if operator == "regex" {
+
+		r, err := regexp.Compile(fmt.Sprintf("%v", value))
+
+		// invalid regex
+		if err != nil {
+			return false, nil
+		}
+
+		match := r.MatchString(fmt.Sprintf("%v", override_value))
+
+		if match {
+			return true, nil
+		} else {
+			return false, nil
 		}
 	}
 
-	if featureFlag.Key == "" {
-		return defaultResult, nil
+	if operator == "not_regex" {
+		var r *regexp.Regexp
+		var err error
+
+		if valueString, ok := value.(string); ok {
+			r, err = regexp.Compile(valueString)
+		} else if valueInt, ok := value.(int); ok {
+			valueString = strconv.Itoa(valueInt)
+			r, err = regexp.Compile(valueString)
+		} else {
+			errMessage := "Regex expression not allowed"
+			return false, errors.New(errMessage)
+		}
+
+		// invalid regex
+		if err != nil {
+			return false, nil
+		}
+
+		var match bool
+		if valueString, ok := override_value.(string); ok {
+			match = r.MatchString(valueString)
+		} else if valueInt, ok := override_value.(int); ok {
+			valueString = strconv.Itoa(valueInt)
+			match = r.MatchString(valueString)
+		} else {
+			errMessage := "Value type not supported"
+			return false, errors.New(errMessage)
+		}
+
+		if !match {
+			return true, nil
+		} else {
+			return false, nil
+		}
 	}
 
-	return poller.getFeatureFlagVariant(featureFlag, key, distinctId)
+	if operator == "gt" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable > valueOrderable, nil
+	}
+
+	if operator == "lt" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable < valueOrderable, nil
+	}
+
+	if operator == "gte" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable >= valueOrderable, nil
+	}
+
+	if operator == "lte" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable <= valueOrderable, nil
+	}
+
+	return false, nil
+
+}
+
+func validateOrderable(firstValue interface{}, secondValue interface{}) (float64, float64, error) {
+	convertedFirstValue, err := interfaceToFloat(firstValue)
+
+	if err != nil {
+		errMessage := "Value 1 is not orderable"
+		return 0, 0, errors.New(errMessage)
+	}
+	convertedSecondValue, err := interfaceToFloat(secondValue)
+
+	if err != nil {
+		errMessage := "Value 2 is not orderable"
+		return 0, 0, errors.New(errMessage)
+	}
+
+	return convertedFirstValue, convertedSecondValue, nil
+
+}
+
+func interfaceToFloat(val interface{}) (float64, error) {
+
+	var i float64
+	switch t := val.(type) {
+	case int:
```

Comment:
just wondering is there any reason we can't consolidate these case statements since we're doing float64(t) to all of them?

Reply:
Not sure if this is different in `go`, but if different cases require the same code, you can put them together like so:

```
case int:
case int8:
case int16:
    i = float64(t)
    return i, nil
```